src/components/boxImage.tsx

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import {
+  StyleSheet,
+  Image,
+  ImageBackground,
+  TouchableOpacity,
+  View,
+  Text,
+} from 'react-native';
+import {IExtraStyles} from '@modules';
+import WhiteCheck from '@assets/White_selected.svg';
+
+interface IBoxImageProps extends IExtraStyles {
+  uri: string;
+  enableTick: boolean;
+  onPress: (item: string) => void;
+  enableOverlay: boolean;
```

Comment:
Overlay be a different component, couple it when required. Other components will ask for it.